brainframe_qt/ui/dialogs/task_configuration/core/zone.py

```diff
@@ -0,0 +1,139 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from typing import List, Optional, Tuple, TypeVar
+
+from brainframe.api import bf_codecs
+from shapely import geometry
+
+T = TypeVar("T")
+
+
+@dataclass
+class Zone(ABC):
```

Comment:
Why not just subclass from the Zone codec dataclass, instead of re-specifying and re-documenting identical fields?

Reply:
It's a fairly common design paradigm to have use-case specific objects in clients to add a bit of separation between the API and client code. For example, the client logic here doesn't care about having stream ids for the zones so the use-cases are a bit different. 

See `Domain` under `Architecture Layers`: https://resocoder.com/2020/03/09/flutter-firebase-ddd-course-1-domain-driven-design-principles/. This is flutter, but should still be applicable.